src/dragAble.js

```diff
@@ -0,0 +1,104 @@
+import { v4 as uuidv4 } from 'uuid';
+/* eslint-disable */
+import { updateTodoStatus } from './todoStatus';
+
+export let todoItem = [];
+  /* eslint-enable */
+
+function reorderArray(from, to, arr) {
+  const newArr = [...arr];
+  const item = newArr.splice(from, 1)[0];
+  newArr.splice(to, 0, item);
+  return newArr;
+}
+
+export function saveTodos() {
+  localStorage.setItem('todos', JSON.stringify(todoItem));
+}
+
+let previousIndex = 0;
+let newIndex = 0;
+
+export const draggableMethods = {
+  onDragStart: (event) => {
+    event.target.classList.add('drag-sort-active');
+    const container = document.getElementById('todo-list');
+    const containerChildren = Array.from(container.children);
+    previousIndex = containerChildren.length - 1 - containerChildren.indexOf(event.target);
+  },
+  onDragEnd: () => {
+  },
+  onDrop: (event) => {
+    event.preventDefault();
+    const container = document.getElementById('todo-list');
+    const draggable = document.querySelector('.drag-sort-active');
+    const containerChildren = Array.from(container.children);
+    newIndex = containerChildren.length - 1 - containerChildren.indexOf(draggable);
+    todoItem = reorderArray(previousIndex, newIndex, todoItem);
+    draggable.classList.remove('drag-sort-active');
+    saveTodos();
+  },
+  onDragOver: (event) => {
+    event.preventDefault();
+    const draggableElements = [...document.querySelectorAll('.draggable:not(.drag-sort-active)')];
+    const draggable = document.querySelector('.drag-sort-active');
+    const container = document.getElementById('todo-list');
+    const afterElement = draggableElements.reduce((closest, child) => {
+      const box = child.getBoundingClientRect();
+      const offset = event.clientY - box.top - box.height / 2;
+      if (offset < 0 && offset > closest.offset) {
+        return { offset, element: child };
+      }
+      return closest;
```

Comment:
- [ ] Currently when I drag a list item still local storage doesn't update a new position when i load the page
- [ ] Need to work on the file that contain methods related to the status updates (completed: true / false). currently when I check a box it doesn't get updated from `true` to `false` or vice versa